agent/lib/kontena/service_pods/creator.rb

```diff
@@ -159,6 +159,7 @@ def ensure_image(name)
       # @param [Docker::Container] service_container
       # @return [Boolean]
       def service_uptodate?(service_container)
+        return false unless service_container.running?
```

Comment:
Yes, the `kontena service stop` will mark the `desired_state: 'stopped'`, and the agent `ServicePodWorker` will execute `Kontena::ServicePods::Stopper`.

Then you run `kontena service start`, the service pod is updated to `desired_state: running`, and the agent `ServicePodWorker` will execute `Kontena::ServicePods::Creator`.... which now destroys the existing stopped container and creates a new one.